ui/html-fragment.reel/html-fragment.js

```diff
@@ -0,0 +1,217 @@
+                    var allowedAttributesForTag, shouldRemoveAttribute,
+                        childAttributes, attribute, attributeName, attributeValue,
+                        delegateResponse, childTagName, child, ii, ll;
+
+                    for (var i = 0, l = children.length; i < l; i++) {
+                        child = children[i];
+                        childTagName = child.tagName.toLowerCase();
+
+                        if (allowedTags && allowedTags.indexOf(childTagName) === -1) {
+                            parent.removeChild(child);
+                            i--;
+                            l--;
+                        } else {
+                            childAttributes = child.attributes;
+                            shouldRemoveAttribute = false;
+                            allowedAttributesForTag = allowedAttributes[childTagName] ||
+                                allowedAttributes['*'];
+                            
+                            for (ii = 0, ll = childAttributes.length; ii < ll; ii++) {
+                                attribute = childAttributes[ii];
+                                attributeName = attribute.name;
+                                attributeValue = attribute.value;
+
+                                if (allowedAttributesForTag &&
+                                    allowedAttributesForTag.indexOf(attributeName) === -1
+                                ) {
+                                    shouldRemoveAttribute = true;
+
+                                    if (attributeName.startsWith('data-') &&
+                                        allowedAttributesForTag.indexOf('data-*') > -1
+                                    ) {
+                                        shouldRemoveAttribute = false;
+                                    }
+                                }
+
+                                if (shouldRemoveAttribute) {
+                                    delegateResponse = this.callDelegateMethod(
+                                        'htmlFragmentWillRemoveAttributeFromNode',
+                                        this,
+                                        attribute,
+                                        child
+                                    );
+
+                                    if (typeof delegateResponse === 'boolean') {
+                                        shouldRemoveAttribute = delegateResponse;
+                                    }
+
+                                    if (shouldRemoveAttribute) {
+                                        child.removeAttribute(attributeName);
+                                        ll--;
+                                        ii--;
+                                    }
+                                } else {
+                                    delegateResponse = this.callDelegateMethod(
+                                        'htmlFragmentWillUseAttributeValueForAttributeNameFromNode',
```

Comment:
let's rename to htmlFragmentWillUseValueForNodeAttribute (this, value, node, attribute)